### ISSUE TITLE
Training Output Problem

### ISSUE THREAD

**Issue by gaopeng-eugene** [zhangliliang/RPN_BF#19]
I1114 13:08:42.275104 31069 sgd_solver.cpp:106] Iteration 150, lr = 5e-05
I1114 13:09:43.542522 31069 solver.cpp:236] Iteration 200, loss = 5.62384
I1114 13:09:43.542560 31069 solver.cpp:252]     Train net output #0: accuracy_1_5x5 = 0.881346
I1114 13:09:43.542567 31069 solver.cpp:252]     Train net output #1: accuracy_1_5x5 = 0.0714286
I1114 13:09:43.542572 31069 solver.cpp:252]     Train net output #2: accuracy_1_7x7 = 0.817905
I1114 13:09:43.542574 31069 solver.cpp:252]     Train net output #3: accuracy_1_7x7 = 0.454545
I1114 13:09:43.542578 31069 solver.cpp:252]     Train net output #4: accuracy_2_5x5 = 0.780908
I1114 13:09:43.542582 31069 solver.cpp:252]     Train net output #5: accuracy_2_5x5 = 1
I1114 13:09:43.542587 31069 solver.cpp:252]     Train net output #6: accuracy_2_7x7 = 0.89241
I1114 13:09:43.542590 31069 solver.cpp:252]     Train net output #7: accuracy_2_7x7 = -1
I1114 13:09:43.542593 31069 solver.cpp:252]     Train net output #8: accuracy_3_5x5 = 0.841772
I1114 13:09:43.542598 31069 solver.cpp:252]     Train net output #9: accuracy_3_5x5 = 0
I1114 13:09:43.542600 31069 solver.cpp:252]     Train net output #10: accuracy_3_7x7 = 0.731013
I1114 13:09:43.542604 31069 solver.cpp:252]     Train net output #11: accuracy_3_7x7 = 1
I1114 13:09:43.542608 31069 solver.cpp:252]     Train net output #12: accuracy_4_5x5 = 0.955696
I1114 13:09:43.542611 31069 solver.cpp:252]     Train net output #13: accuracy_4_5x5 = 0.5
I1114 13:09:43.542628 31069 solver.cpp:252]     Train net output #14: boxiou_1_5x5 = 0.596698
I1114 13:09:43.542631 31069 solver.cpp:252]     Train net output #15: boxiou_1_7x7 = 0.54248
I1114 13:09:43.542635 31069 solver.cpp:252]     Train net output #16: boxiou_2_5x5 = 0.59975
I1114 13:09:43.542639 31069 solver.cpp:252]     Train net output #17: boxiou_2_7x7 = -1
I1114 13:09:43.542642 31069 solver.cpp:252]     Train net output #18: boxiou_3_5x5 = 0.547801
I1114 13:09:43.542646 31069 solver.cpp:252]     Train net output #19: boxiou_3_7x7 = 0.58118
I1114 13:09:43.542650 31069 solver.cpp:252]     Train net output #20: boxiou_4_5x5 = 0.600571
I1114 13:09:43.542675 31069 solver.cpp:252]     Train net output #21: loss_1_5x5 = 1.26751 (* 0.9 = 1.14076 loss)
I1114 13:09:43.542682 31069 solver.cpp:252]     Train net output #22: loss_1_5x5 = 0.000743121 (* 0.9 = 0.000668809 loss)
I1114 13:09:43.542688 31069 solver.cpp:252]     Train net output #23: loss_1_7x7 = 3.04133 (* 0.9 = 2.7372 loss)
I1114 13:09:43.542695 31069 solver.cpp:252]     Train net output #24: loss_1_7x7 = 0.000900266 (* 0.9 = 0.00081024 loss)
I1114 13:09:43.542701 31069 solver.cpp:252]     Train net output #25: loss_2_5x5 = 0.374449 (* 1 = 0.374449 loss)
I1114 13:09:43.542706 31069 solver.cpp:252]     Train net output #26: loss_2_5x5 = 0.000737343 (* 1 = 0.000737343 loss)
I1114 13:09:43.542711 31069 solver.cpp:252]     Train net output #27: loss_2_7x7 = 0.183456 (* 1 = 0.183456 loss)
I1114 13:09:43.542717 31069 solver.cpp:252]     Train net output #28: loss_2_7x7 = 0 (* 1 = 0 loss)
I1114 13:09:43.542722 31069 solver.cpp:252]     Train net output #29: loss_3_5x5 = 0.471607 (* 1 = 0.471607 loss)
I1114 13:09:43.542728 31069 solver.cpp:252]     Train net output #30: loss_3_5x5 = 0.00156796 (* 1 = 0.00156796 loss)
I1114 13:09:43.542733 31069 solver.cpp:252]     Train net output #31: loss_3_7x7 = 0.351334 (* 1 = 0.351334 loss)
I1114 13:09:43.542739 31069 solver.cpp:252]     Train net output #32: loss_3_7x7 = 0.000982699 (* 1 = 0.000982699 loss)
I1114 13:09:43.542745 31069 solver.cpp:252]     Train net output #33: loss_4_5x5 = 0.359949 (* 1 = 0.359949 loss)
I1114 13:09:43.542750 31069 solver.cpp:252]     Train net output #34: loss_4_5x5 = 0.000321437 (* 1 = 0.000321437 loss)

**Comment by gaopeng-eugene** [comment]
Thank you so much for sharing the code. I can run your code successfully. However, I am confused about the training output. Why there are two loss_1_5_5 and accuracy_1_5_5? From my understanding about your prototxt, there should be only one output? 

Can you clarify that?

**Comment by gaopeng-eugene** [comment]
Another Problem about the network architecture. Why the cls_num is 5? In my opinion, your RPN network is doing binary classification?

**Comment by zhangliliang** [comment]
The attached log might be produced by the MS-CNN repo, but not the RPN+BF repo.
